nixos/modules/services/networking/knot.nix

```diff
@@ -0,0 +1,93 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.knot;
+
+  configFile = pkgs.writeText "knot.conf" cfg.extraConfig;
+  socketFile = "/run/knot/knot.sock";
+
+  knotConfCheck = file: pkgs.runCommand "knot-config-checked"
+    { buildInputs = [ cfg.package ]; } ''
+    ln -s ${configFile} $out
+    knotc --config=${configFile} conf-check
+  '';
+
+  knot-cli-wrappers = pkgs.stdenv.mkDerivation {
+    name = "knot-cli-wrappers";
+    buildInputs = [ pkgs.makeWrapper ];
+    buildCommand = ''
+      mkdir -p $out/bin
+      makeWrapper ${cfg.package}/bin/knotc "$out/bin/knotc" \
+        --add-flags "--config=${configFile}" \
+        --add-flags "--socket=${socketFile}"
+      makeWrapper ${cfg.package}/bin/keymgr "$out/bin/keymgr" \
+        --add-flags "--config=${configFile}"
+      for executable in kdig khost kjournalprint knsec3hash knsupdate kzonecheck
+      do
+        ln -s "${cfg.package}/bin/$executable" "$out/bin/$executable"
+      done
```

Comment:
Man pages: perhaps I'd simply add this here:
```bash
mkdir -p "$out/share"
ln -s '${knot-dns.bin}/share/man' "$out/share/"
```
The closure remains the same anyway.

Reply:
Actually, what we could additionally build is the html docs, i.e. offline version of https://www.knot-dns.cz/documentation/  I can modify the `knot-dns` expression to produce those, if you'd find them useful.

Reply:
I've actually always viewed more extensive documentation on project websites. Especially since servers usually run headless and are configured from machines that have access to a browser.

Reply:
Right... it would be more like installing the docs (without the server) on the dev machine while having just the binaries (without docs) on the server.  I suppose people are used to online docs nowadays.

Reply:
I added a link to the version-specific HTML documentation to the systemd units `Documentation=` key.